feat: first updated esds communication KIT. (#1555)
eclipse-tractusx/eclipse-tractusx.github.io: docs-kits/kits/esdscom-kit/development-view/development-view.md

The connectors act as trusted, policy-enforced gateways on both sides, ensuring that data is only shared with verified partners and under agreed terms.
Each notification consists of a header and a body.

The header contains the routing and identification information:

- the BPNL (Business Partner Number Legal) of both the sending and receiving company
- a unique message ID assigned to each individual message, and a related message ID that references the original request. The related message ID is particularly important when a supplier sends multiple separate responses to a single request, for example when emission data for different operators or goods is compiled and returned in stages. Each response can be matched back to the originating request via this identifier.

The following data fields in the eSDScom data model allow an unique assignment of the documents:

- Supplier DUNS number
- Formulation (specification) number
- Version number of the SDS (current and previous)
- Date of compilation
- Supplier product name
- Country (legal area)
- Language

### Data Schema

#### Semantic Models

Model: eSDScom
Version: 1.0.0

> Namespace: urn:samm:io.catenax.eSDScom:1.0.0

Reviewer: ```suggestion
SemanticId: `urn:samm:io.catenax.esdscom:1.0.0#eSDScom`
 
```

> 

Reviewer: ```suggestion
[`https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.esdscom/1.0.0/eSDScom.ttl`](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.esdscom/1.0.0/eSDScom.ttl)
```

The connectors act as trusted, policy-enforced gateways on both sides, ensuring that data is only shared with verified partners and under agreed terms.
Each notification consists of a header and a body.

The header contains the routing and identification information:

- the BPNL (Business Partner Number Legal) of both the sending and receiving company
- a unique message ID assigned to each individual message, and a related message ID that references the original request. The related message ID is particularly important when a supplier sends multiple separate responses to a single request, for example when emission data for different operators or goods is compiled and returned in stages. Each response can be matched back to the originating request via this identifier.

The following data fields in the eSDScom data model allow an unique assignment of the documents:

- Supplier DUNS number
- Formulation (specification) number
- Version number of the SDS (current and previous)
- Date of compilation
- Supplier product name
- Country (legal area)
- Language

### Data Schema

#### Semantic Models

Model: eSDScom
Version: 1.0.0
SemanticId: `urn:samm:io.catenax.esdscom:1.0.0#eSDScom`
 

> [`https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.esdscom/1.0.0/eSDScom.ttl`](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.esdscom/1.0.0/eSDScom.ttl)

Reviewer: ```suggestion
| Model Type | Link |
|-------------|-----|
| urn:samm:io.catenax.esdscom:1.0.0#eSDScom (.ttl) | [https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.esdscom/1.0.0/eSDScom.ttl](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.esdscom/1.0.0/eSDScom.ttl) |
| JSON Payload | [https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.esdscom/1.0.0/gen/eSDScom.json](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.esdscom/1.0.0/gen/eSDScom.json) |
| JSON Schema | [https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.esdscom/1.0.0/gen/eSDScom-schema.json](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.esdscom/1.0.0/gen/eSDScom-schema.json) |
```

> The following diagram shows the digital twin structure used to represent eSDScom data in the Catena-X dataspace.

Reviewer: ```suggestion

The following diagram shows the digital twin structure used to represent eSDScom data in the Catena-X dataspace.
```

which is available at
https://creativecommons.org/licenses/by/4.0/legalcode.

SPDX-License-Identifier: CC-BY-4.0
-->

import Kit3DLogo from '@site/src/components/2.0/Kit3DLogo';

<Kit3DLogo kitId="esdscom" />

## Development View

eSDScom is a data model, not a software. No API other than the digital twin registry APIs.
This document provides a developer-focused overview of the eSDScom KIT, including a technical breakdown of the architecture, and data exchange protocols and semantic models.

### Target Audience

Software Developers, Solution Architects, Technical Leads, API Developers, Integration Engineers.

### Focus

Technical implementation, APIs, and development resources

Required contents in the following sub-chapters:

### Architecture Overview

This section is non-normative.

The architecture of this KIT is based on Catena-X standards.

To participate in the eSDScom use case, the following single standards MUST be fulfilled by all participants for which the standard is relevant:

- CX-0002 Digital Twins in Catena-X
- CX–0018 Sovereign Data Exchange

To participate in the eSDScom use case, the following single standard MUST be fulfilled only by data providers:

> - CX-XXXX eSDScom In the Catena-X technical landscape, the standard assumes the use of CX-0126-IndustryCorePartType.

Reviewer: ```suggestion
- CX-0162 eSDScom In the Catena-X technical landscape, the standard assumes the use of CX-0126-IndustryCorePartType.
```

The architecture follows a pull-based pattern: data is pushed from supplier to consumer (or other way around) but made accessible by the data provider. This standard applies equally to Data Providers and Consumers. All parties involved in the exchange must implement the agreed data structure and access patterns to ensure interoperability within the Catena-X network.

> 1. All Identification and Access Management (IAM) follows the CX-0015 IAM & Access Control Paradigm for users and Clients standard [https://catenax-ev.github.io/docs/standards/CX-0015-IAMandAccessControlParadigmForUsersAndClients] which lays out the principles for both, data providers and data consumers.

Reviewer: Just a quick modification that I have just saw which is wrong:

```suggestion
1. All Data Exchange Requirements follows the CX-0018 Dataspace Connectivity [https://catenax-ev.github.io/docs/standards/CX-0018-DataspaceConnectivity] and which lays out the principles for both, data providers and data consumers. More information can be found in the [Connector KIT](https://eclipse-tractusx.github.io/docs-kits/kits/connector-kit/adoption-view).
```

The IAM principal works different in Catena-X it is only valid for the portal, not for the data exchange, so we are not authenticating using Keycloak the data exchange, but rather using the CX-0018 standard which is core.

2. Both, the data providers and consumers shall be identified in the Catena-X dataspace following the CX-0010 Business Partner Number Standard [https://catenax-ev.github.io/docs/standards/CX-0010-BusinessPartnerNumber] so that a clear identification of both parties enables automatic data exchange.
The Onboarding of the parties on the Catena-X Dataspace shall be effectuated following the CX-0006 Registration and initial Onboarding standard [https://catenax-ev.github.io/docs/standards/CX-0006-RegistrationAndInitialOnboarding].

> 

Reviewer: ```suggestion
3. The data provider party must have a digital twin registry which comply with the [CX-0002](https://catenax-ev.github.io/docs/standards/CX-0002-DigitalTwinsInCatenaX) standard, detailed in the [Digital Twin KIT](https://eclipse-tractusx.github.io/docs-kits/kits/digital-twin-kit/adoption-view). In this way the eSDScom data model can be provided in a AAS Shell Descriptor as a Submodel, more information in the [development view](./development-view/development-view.md).
```

> 3. The data provider party must have a digital twin registry which comply with the [CX-0002](https://catenax-ev.github.io/docs/standards/CX-0002-DigitalTwinsInCatenaX) standard, detailed in the [Digital Twin KIT](https://eclipse-tractusx.github.io/docs-kits/kits/digital-twin-kit/adoption-view). In this way the eSDScom data model can be provided in a AAS Shell Descriptor as a Submodel, more information in the [development view](./development-view/development-view.md).

Reviewer: ```suggestion
3. The data provider party must have a digital twin registry which comply with the [CX-0002](https://catenax-ev.github.io/docs/standards/CX-0002-DigitalTwinsInCatenaX) standard, detailed in the [Digital Twin KIT](https://eclipse-tractusx.github.io/docs-kits/kits/digital-twin-kit/adoption-view). In this way the eSDScom data model can be provided in a AAS Shell Descriptor as a Submodel, more information can be found in the architecture diagram below.
```

The eSDScom standard defines a structured data model that enables the exchange of SDS between suppliers and OEMs in the automotive industry. It is designed to reduce the current manual effort involved in gathering and processing data needed for occupational workplace and environmental safety, which today is commonly exchanged via e-Mail and PDF files.
This standard is not embedded in a broader Catena-X use case or initiative but operates as a standalone specification that supports the bilateral exchange of information. It is applicable whenever a company needs to obtain SDS data.

System Architecture [High-level architecture diagram and explanation] Architecture Principles:

- Modularity: Loosely coupled components
- Scalability: Horizontal scaling support
- Security: End-to-end encryption
- Interoperability: Standards-based APIs
- Observability: Built-in monitoring and logging

> The architecture of this KIT is based on Catena-X standards. First of all Identification and Access Management (IAM) follows the CX-0015 IAM & Access Control Paradigm for users and Clients standard [https://catenax-ev.github.io/docs/standards/CX-0015-IAMandAccessControlParadigmForUsersAndClients] which lays out the principles for both, data providers and data consumers.

Reviewer: ```suggestion
The architecture of this KIT is based on Catena-X standards. First all Data Exchange Requirements follows the CX-0018 Dataspace Connectivity [https://catenax-ev.github.io/docs/standards/CX-0018-DataspaceConnectivity] and which lays out the principles for both, data providers and data consumers.
```

---
id: development-view
title: Development View
description: Overview on the eSDScom KIT
sidebar_position: 3
---

<!--

> Copyright(c) 2025 Contributors to the Eclipse Foundation

Reviewer: ```suggestion
Copyright(c) 2026 Contributors to the Eclipse Foundation
```

The connectors act as trusted, policy-enforced gateways on both sides, ensuring that data is only shared with verified partners and under agreed terms.
Each notification consists of a header and a body.

The header contains the routing and identification information:

- the BPNL (Business Partner Number Legal) of both the sending and receiving company
- a unique message ID assigned to each individual message, and a related message ID that references the original request. The related message ID is particularly important when a supplier sends multiple separate responses to a single request, for example when emission data for different operators or goods is compiled and returned in stages. Each response can be matched back to the originating request via this identifier.

The following data fields in the eSDScom data model allow an unique assignment of the documents:

- Supplier DUNS number
- Formulation (specification) number
- Version number of the SDS (current and previous)
- Date of compilation
- Supplier product name
- Country (legal area)
- Language

### Data Schema

#### Semantic Models

Model: eSDScom
Version: 1.0.0

> Namespace: urn:samm:io.catenax.eSDScom:1.0.0

Reviewer: @jSchuetz88 could you check here please.

The following diagram shows the digital twin structure used to represent eSDScom data in the Catena-X dataspace.

![eSDScom Digital Twin Structure](../resources/digital-twin-structure.svg)

##### Description

Purpose: The SAMM defines the eSDScom data model used to exchange eSDScom-relevant SDS information between an OEM (customer) and a supplier.
Request vs Response separation: The SAMM cleanly separates request and response concerns: the request model specifies what the OEM may ask for (scope, requested elements, identifiers), while the response model specifies what the supplier must provide the requested SDS data, enabling clear responsibilities and automated schema generation.

#### Standards

Technical standards and protocol compliance are not applicable.

#### Tutorials

> Developer quick-start guides and code examples are not yet available.

Reviewer: ```suggestion
```

I would take this out, because it does not adds value.
